src/aarch64/logic-aarch64.cc

```diff
@@ -168,11 +168,12 @@ SimFloat16 Simulator::UFixedToFloat16(uint64_t src,
 
 
 uint64_t Simulator::GenerateRandomTag(uint16_t exclude) {
-  uint64_t rtag = nrand48(rand_state_) >> 28;
+  // Generate a positive signed 4 bit integer from a 48bit random number
+  uint64_t rtag = rand_gen_() >> 45;
   VIXL_ASSERT(IsUint4(rtag));
```

Comment:
If I understand linear_congruential_engine correctly, the result of this is a 3-bit rather than 4-bit integer, isn't it?

Reply:
Yeah, would give you an unsigned 3bit number.

This is the same range of values as a 4bit signed int limited to only positive values, as the sign bit would be zero.

I wasn't exactly clear on what exactly was expected here, but prior code took a positive signed 32bit number from nrand48, and shifted into a 4bit number.
With the sign bit always being zero, you would only end up with 3bits of random values before aswell.
I match this result by shifting past the sign bit.

If the intension is to use all 4bits, I can make that change? Or would it be a case of correcting the comment I added?

Reply:
Hmm, looking at this more closely, I think the original code is wrong. It should be picking a random tag between 0 and 15 inclusive, but is only picking values up to 7 (and rarely, 8). The test doesn't notice this either, for subtle reasons.

We can either take what you have here, and I'll modify it later to correct the range, or you can try to fix the range to be the proper 0 to 15 inclusive yourself. The relevant test will pass either way, so that'll need to be patched later - I have a fix locally.

Reply:
I can make the change later today
As for exclude (just below), what range of values does that need to be? The same as rtag (0-15)? Or a larger range?

Reply:
Okay